tests/Aspire.Hosting.Tests/DistributedApplicationTests.cs

```diff
@@ -334,6 +334,55 @@ await foreach (var (_, r) in kubernetes!.WatchAsync<T>(cancellationToken: cancel
         await app.StopAsync();
     }
 
+    [LocalOnlyFact("docker")]
+    public async Task ContainerHostPortDoesNotConflictWithServiceHostPort()
+    {
+        var hostPort = 6380;
+
+        var testProgram = CreateTestProgram();
+
+        testProgram.AppBuilder.Services.AddLogging(b => b.AddXunit(_testOutputHelper));
+
+        testProgram.AppBuilder.AddContainer("redis0", "redis")
+            .WithServiceBinding(containerPort: 6380, hostPort: hostPort, name: "tcp", env: "REDIS_PORT");
+
+        await using var app = testProgram.Build();
+
+        var kubernetes = app.Services.GetRequiredService<KubernetesService>();
+
+        await app.StartAsync();
+
+        async Task<T> GetResourceByNameAsync<T>(string resourceName, Func<T, bool> ready, CancellationToken cancellationToken) where T : CustomResource
```

Comment:
nit: move to the bottom of the method (purely a stylistic preference so if you have stronger feelings feel free to ignore). Also the name is a little confusing, took me a moment to figure out what the method was doing.

Reply:
I wrote this code, so that's my fault 😊 . I think we should do a big push to get these tests running on the CI in p3